MPI_rot.f90

```diff
@@ -1,55 +1,67 @@
 module mpi_rot
   use globalvar,only:total_prc,my_rank,nvar_h,nvar_m,mpi_siz,mpi_pos,margin,&
        ix,jx,kx,ndim,flag_mpi_split,neighbor,U_h,U_m,flag_mhd,flag_pip,&
-       cno,vmpi,flag_mpi,flag_bnd
+       cno,vmpi,flag_mpi,flag_bnd, &
+       dimsFile, ig
   implicit none
   include "mpif.h"
   integer ierr
   integer status(6)
 contains
-  subroutine set_mpi    
+  subroutine set_mpi
+    integer i
     !charm for MPI
     call mpi_init(ierr)
     call mpi_comm_size(MPI_COMM_WORLD,total_prc,ierr)
-    call mpi_comm_rank(MPI_COMM_WORLD,my_rank,ierr)   
- 
+    call mpi_comm_rank(MPI_COMM_WORLD,my_rank,ierr)
+
     if(flag_mpi.eq.0 .or.my_rank.eq.0) print *,"total cpu",total_prc
     vmpi=nvar_h+nvar_m
     write(cno,"(i4.4)")my_rank
-!   print *,"TOT",total_prc,my_rank,cno 
-   if(total_prc.eq.1) then
-       mpi_siz(:)=1
-       mpi_pos(:)=0       
+    !   print *,"TOT",total_prc,my_rank,cno
+    if(total_prc.eq.1) then
+      mpi_siz(:)=1
+      mpi_pos(:)=0
     else
-       call set_mpi_pos
+      call set_mpi_pos
     endif
-    ix=(ix-2*margin(1))/mpi_siz(1)+2*margin(1)
-    jx=(jx-2*margin(2))/mpi_siz(2)+2*margin(2)
-    kx=(kx-2*margin(3))/mpi_siz(3)+2*margin(3)    
+
+    ! store full ranges + margins for use in write-IO
```

Comment:
This is the set of changes that fixes the uneven grid-split **bug**. The bug comes from the original code doing integer division ala `1000/3=333`. The remainder was never accounted for. The solution is to take the remained and add it to the last cell block along the relevant axis.

I also stored the full grid + margins range in `dimsFile` for use in writing and reading HDF5 data.